src/Creator.tsx

```diff
@@ -0,0 +1,234 @@
+import React, { useContext, useEffect, useMemo, useState } from 'react';
+import YAML from 'yaml';
+import {
+  Button,
+  Flex,
+  FlexItem,
+  Grid,
+  GridItem,
+  PageGroup,
+  PageSection,
+  Title,
+} from '@patternfly/react-core';
+import './Creator.scss';
+import './components/CatalogSection.scss';
+import {
+  AllQuickStartStates,
+  QuickStart,
+  QuickStartContext,
+  QuickStartDrawer,
+  QuickStartStatus,
+  useValuesForQuickStartContext,
+} from '@patternfly/quickstarts';
+import WrappedQuickStartTile from './components/WrappedQuickStartTile';
+import CreatorWizard, {
+  CreatorWizardState,
+  EMPTY_TASK,
+} from './components/creator/CreatorWizard';
+import { itemKindMeta } from './components/creator/meta';
+import { AppContext } from './AppContext';
+
+function makeDemoQuickStart(state: CreatorWizardState): QuickStart {
+  const selectedTypeMeta =
+    state.type !== null ? itemKindMeta[state.type] : null;
+
+  return {
+    metadata: {
+      name: 'test-quickstart',
+      kind: 'QuickStarts',
+      ...(selectedTypeMeta?.extraMetadata ?? {}),
+    },
+    spec: {
+      displayName: state.title,
+      icon: null,
+      description: state.description,
+      introduction:
+        selectedTypeMeta?.hasTasks === true ? state.introduction : undefined,
+      prerequisites:
+        selectedTypeMeta?.hasTasks === true ? state.prerequisites : undefined,
+      link:
+        selectedTypeMeta?.fields?.url === true
+          ? {
+              href: state.url,
+              text: 'View documentation',
+            }
+          : undefined,
+      type:
+        selectedTypeMeta !== null
+          ? {
+              text: selectedTypeMeta.displayName,
+              color: selectedTypeMeta.tagColor,
+            }
+          : undefined,
+      durationMinutes:
+        selectedTypeMeta?.fields?.duration === true && state.duration > 0
+          ? state.duration
+          : undefined,
+      tasks:
+        selectedTypeMeta?.hasTasks === true
+          ? state.tasks.map((task) => {
+              try {
+                return {
+                  ...YAML.parse(task.yamlContent),
+                  title: task.title,
+                };
+              } catch (e) {
+                return {
+                  title: task.title,
+                  description: 'An error occurred while parsing the task.',
+                };
+              }
+            })
+          : undefined,
+    },
+  };
+}
+
+const Creator = () => {
+  const { onNavigate } = useContext(AppContext);
+
+  const [state, setState] = useState<CreatorWizardState>({
```

Comment:
I have the separate state type so that we can preserve information that shouldn't be preserved in the quickstart. For instance, we can preserve the duration or tasks if the user goes back and changes the type of the quickstart.

I'm not sure what would be gained by breaking the state out into separate variables? We would still have to re-compute the quickstart when it changes.

Reply:
> I have the separate state type so that we can preserve information that shouldn't be preserved in the quickstart.

Then have one for the quickstart and one for the other stuff that is separate. There is no logical difference. The format of the quickstart is not so bad on its own that it has to be mapped to something else.